api/api.go

```diff
@@ -0,0 +1,81 @@
+package api
+
+import (
+	"context"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"github.com/spiffe/spike-sdk-go/api/entity/data"
+	"github.com/spiffe/spike-sdk-go/spiffe"
+)
+
+type Api struct {
+	source *workloadapi.X509Source
+}
+
+func New() *Api {
+	defaultEndpointSocket := spiffe.EndpointSocket()
+
+	source, _, err := spiffe.Source(context.Background(), defaultEndpointSocket)
+	if err != nil {
+		return nil
+	}
+
+	return &Api{source: source}
+}
```

Comment:
It's useful to encapsulate the `source` in a constructor so that we don't have to pass to every single method.